src/conditional.jl

```diff
@@ -195,11 +195,11 @@ function grad_apply_sparsity!(rbm::ConditionalRBM, X::Mat,
                   throw(ArgumentError("If :sparsity_cost is used, :sparsity_target should also be defined")))
     curr_sparsity = mean(hid_means(rbm, vis))
     penalty = cost * (curr_sparsity - target)
-    axpy!(-penalty, dW, dW)
-    axpy!(-penalty, dA, dA)
-    axpy!(-penalty, dB, dB)
-    axpy!(-penalty, db, db)
-    axpy!(-penalty, dc, dc)
+    axpy!(-penalty, ones(size(dW)), dW)
```

Comment:
`ones()` should take into account type of RBM, e.g. `ones(T, size(dW))` instead of `ones(size(dW))`. Also, it would be nice to compare it to using ordinary `dW -= penalty`, because creating additional matrix of ones has its own cost as well.